### PR TITLE
Add agent mode identifier

### PR DESCRIPTION
**Pull request by SuperQ**
Identify in the logs and liveness endpoints if the server is running in
Agent mode or not.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Review by roidelapluie — COMMENTED** [review]
While we are at it, should we have a metric?

File: cmd/prometheus/main.go [hunk 1]
```diff
@@ -531,7 +531,12 @@ func main() {
 	klogv2.ClampLevel(6)
 	klogv2.SetLogger(log.With(logger, "component", "k8s_client_runtime"))
 
-	level.Info(logger).Log("msg", "Starting Prometheus", "version", version.Info())
+	modeAppName := "Prometheus"
+	if agentMode {
+		modeAppName += " Agent"
+	}
+
+	level.Info(logger).Log("msg", "Starting " + modeAppName, "version", version.Info())
```
Comment (roidelapluie) [inline cmd/prometheus/main.go:539]: should we have mode=server or mode=agent as a field instead?

**Comment by roidelapluie** [comment]
friendly ping

**Comment by SuperQ** [comment]
Sorry, forgot about this one. Added the `mode` field to the log line.

**Comment by SuperQ** [comment]
Ping @roidelapluie

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**